### PR TITLE
Switch data export script to python and provide data dictionary alongside input fields

### PR DESCRIPTION
**Pull request by attwad**
This is to make the data dictionary part of the export process to make sure we are always in sync (not provide fields we do not document).

The fields are now described in a yaml file, so I switched to python that just invokes the mongoexport CLI using subprocess, no big change overall.

For #780

### CODE REVIEW

**Review by allysonjp715 — APPROVED** [review]

File: .github/workflows/case-data-export-prod.yml [hunk 1]
```diff
@@ -28,11 +28,17 @@ jobs:
           sudo dpkg --purge mongodb-org-database-tools-extra
           sudo apt-get update
           sudo apt-get install -f -y --allow-downgrades mongodb-org=4.2.6 mongodb-org-server=4.2.6 mongodb-org-shell=4.2.6 mongodb-org-mongos=4.2.6 mongodb-org-tools=4.2.6 libcurl3
-
+      - uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
+      - name: Install python dependencies
+        run: |
+          python -m pip install --upgrade pip
```
Comment (axmb) [inline .github/workflows/case-data-export-prod.yml:36]: Any reason we're using `python` here and `python3` below? Mostly a nit -- they're probably symlinked to the same thing in `setup-python`.

File: .github/workflows/case-data-export-prod.yml [hunk 2]
```diff
@@ -28,11 +28,17 @@ jobs:
           sudo dpkg --purge mongodb-org-database-tools-extra
           sudo apt-get update
           sudo apt-get install -f -y --allow-downgrades mongodb-org=4.2.6 mongodb-org-server=4.2.6 mongodb-org-shell=4.2.6 mongodb-org-mongos=4.2.6 mongodb-org-tools=4.2.6 libcurl3
-
+      - uses: actions/setup-python@v1
```
Comment (axmb) [inline .github/workflows/case-data-export-prod.yml:31]: I think v2 is the latest GA hotness.

File: .github/workflows/case-data-export-prod.yml (hunk 1 above)
Comment (attwad) [reply .github/workflows/case-data-export-prod.yml:36]: ah I didnt notice, switched to pythn3 for consistency.

File: .github/workflows/case-data-export-prod.yml (hunk 2 above)
Comment (attwad) [reply .github/workflows/case-data-export-prod.yml]: cool, done

**Pull request merged**